pkg/core/app/app.go

```diff
@@ -464,8 +466,15 @@ func (a *App) run(ctx context.Context) models.AppError {
 	select {
 	case <-ctx.Done():
 		a.logger.Debug("context cancelled, error while waiting for the app to exit", zap.Error(ctx.Err()))
-		return models.AppError{AppErrorType: models.ErrCtxCanceled, Err: nil}
+		return models.AppError{AppErrorType: models.ErrCtxCanceled, Err: ctx.Err()}
 	default:
+		if a.Mode == models.MODE_RECORD && a.EnableTesting {
+			a.logger.Info("waiting for some time before returning the error to allow recording of test cases when testing keploy with itself")
+			time.Sleep(3 * time.Second)
```

Comment:
these delays may vary from user's machine so need find a way to do this without dealys

Reply:
Yes, I agree but in record mode, we anyway wait for 2 seconds after the last test case. The other way is that we can read the pre-recorded test cases of a particular test-set and find out how many test cases were there and wait till all the test cases are made. But this also require a lot of unnecessary code duplication. We can improve in the future version. Since I've already added this point in the caveats. 
In this PR I've not covered the caveats. Hence the mention.